pikaur/build.py

```diff
@@ -333,12 +337,20 @@ def _set_built_package_path(self):
             self.built_package_path = built_package_path
 
     def build(self, args, all_package_builds):
-        repo_path = self.repo_path
         if os.path.exists(self.build_dir):
             remove_dir(self.build_dir)
-        shutil.copytree(repo_path, self.build_dir)
 
-        src_info = SrcInfo(repo_path, self.package_name)
+        mkdir_command = ['mkdir', '-p', os.path.dirname(self.build_dir)]
```

Comment:
i think it also could be something like:

```python
isolated_process = subprocess.Popep(isolate_root_cmd(['sleep', '99']))
shutil.copytree(self.repo_path, self.build_dir)
isolated_process.kill()
```

but i am not sure if it will be any better from the point of style if it will work

Reply:
We could write a context manager that enters in a "dynamic user context" by opening a bash -c read on __enter__ and writing a "\n" byte + .wait(). But yeah, it's ugly. I'd like to see what the systemd people think of that.

Reply:
actually if that solution it can be writen as a context manager so look quite nicely in the code

Reply:
I think it would look like this (untested):

```
@contextlib.contextmanager
def dynuser_context():
    if running_as_root():
        p = subprocess.Popen(isolate_root_cmd(['sh', '-c', 'read']))
        yield
        p.kill()
    else:
        yield
```

and then:

```
with dynuser_context():
    shutil.copytree(self.repo_path, self.build_dir)
```

Reply:
so if it's working as expected that looks fine for me